subsys/net/ip/Kconfig

```diff
@@ -279,6 +279,22 @@ config NET_TC_NUM_PRIORITIES
 	default NUM_COOP_PRIORITIES if NET_TC_THREAD_COOPERATIVE
 	default NUM_PREEMPT_PRIORITIES if NET_TC_THREAD_PREEMPTIVE
 
+config NET_TC_THREAD_PRIO_CUSTOM
+	bool "Net threads priority custom"
+	default n
+	help
+	  Customise net threads priority by each.
+
+if NET_TC_THREAD_PRIO_CUSTOM
+config NET_TC_TX_THREAD_BASE_PRIO
+	int "Net TC TX thread priority base"
+	default 0
```

Comment:
It looks like there is now possibility that the user is able to set the base priority so that the the tc thread priorities do not "fit" in the priority range. There are some build time checks in the code so this might not be an issue, but could you verify that?
What I mean by this is that by default `CONFIG_NUM_PREEMPT_PRIORITIES` is 15 and `CONFIG_NUM_COOP_PRIORITIES` is 16. If user has selected `CONFIG_NET_TC_THREAD_COOPERATIVE` (which is the default) and `CONFIG_NET_TC_TX_COUNT=8` and if you select the base priority to be for example 12, then the priority range would "overflow" by 4 units. We should not allow this and code tries to verify that. So could you verify that all is working well with these kind of special cases.

Reply:
I verified this case and get following assert on board. it it expected?

ASSERTION FAIL [((((prio)) == 15 && z_is_idle_thread_entry((entry))) || (((15 - 1) >= ((-16))) && ((prio)) >= ((-16)) && ((prio)) <= (15 - 1)))] @ WEST_TOPDIR/zephyr/kernel/thread.c:601
        invalid priority (239); allowed range: 14 to -16

Reply:
Yes, that is expected, thanks for verifying this.